Prevent fire from spreading onto nonflammable tiles

anothersimulacrum opened the pull request:
#### Summary
SUMMARY: Bugfixes "Prevent fire from spreading through non-flammable walls"

#### Purpose of change
Fire in labs and refugee centers (and I'm sure plenty of other locations) could leap through the non-flammable walls and set other rooms on fire. This stops that.

#### Describe the solution
Add a new boolean `can_burn` to `map::proccess_fields_in_submap`, that checks if a tile has the `NONFLAMMABLE` or `FIRE_CONTAINER` flags, and then uses that to determine whether or not to burn the tile it is on.
Swap places where `can_spread` was used for this to `can_burn`.
Add check before fire spreading that the tile it is spreading onto can burn.

#### Describe alternatives you've considered
Considered adding new flags to make it so fire could not spread off walls, but that was complicated, and accomplished what I wanted in a very roundabout way. This is a much better solution that that.

#### Additional context
I will add a complete list of all terrains with this flag tomorrow.
Some before and after gifs of fire with this.

![pre](https://user-images.githubusercontent.com/42699974/60703716-04202300-9ef2-11e9-84b0-ce0cf445b9ab.gif)
![post](https://user-images.githubusercontent.com/42699974/60703715-04202300-9ef2-11e9-9b59-ace6fbee9435.gif)

![pre1](https://user-images.githubusercontent.com/42699974/60703758-18fcb680-9ef2-11e9-92a4-087e55edef9d.gif)
![post1](https://user-images.githubusercontent.com/42699974/60703763-1b5f1080-9ef2-11e9-9089-6872e104def4.gif)

There should be very little to no impact in this house burning down because of these changes.
![prehouse](https://user-images.githubusercontent.com/42699974/60703766-1e5a0100-9ef2-11e9-8073-3cc53da4c8d8.gif)
![posthouse](https://user-images.githubusercontent.com/42699974/60703769-20bc5b00-9ef2-11e9-8cbb-31121468753c.gif)

BevapDin commented:
So, how do we call things that have both the `"FLAMMABLE"` and the `NONFLAMMABLE`" tag? And how do we call things that have neither flags?

Isn't the existing tag enough?

BevapDin commented on src/mapdata.cpp:
```diff
@@ -139,6 +139,7 @@ static const std::unordered_map<std::string, ter_bitflags> ter_bitflags_map = {
         { "ALLOW_FIELD_EFFECT",       TFLAG_ALLOW_FIELD_EFFECT }, // Fire, acid
         { "COLLAPSES",                TFLAG_COLLAPSES },      // building "remodeling"
         { "FLAMMABLE",                TFLAG_FLAMMABLE },      // fire bad! fire SLOW!
+        { "NONFLAMMABLE",             TFLAG_NONFLAMMABLE },   // fire
```
What does the comment mean?

anothersimulacrum replied:
It's not relevant anymore, but the comment meant that that was added to that because it was used in fire calculations, like the other flags in there for fire calculations.
But I see how it was unclear, and should have been made more clear.

anothersimulacrum commented:
Update: I've removed the `NONFLAMMABLE` tag, and added a new function `is_flammable` that returns true if the terrain or furniture has any of the `FLAMMABLE`, `FLAMMABLE_ASH`, or `FLAMMABLE_HARD` flags.

Here are updated post-gifs.
![postcrate](https://user-images.githubusercontent.com/42699974/60743770-8811f380-9f62-11e9-9c42-868e35bac113.gif)
![postfirehouse](https://user-images.githubusercontent.com/42699974/60743771-8811f380-9f62-11e9-8b87-ed68fdae0d9c.gif)
![posthouse2](https://user-images.githubusercontent.com/42699974/60743772-8811f380-9f62-11e9-933f-d37c28a611c7.gif)
